### PR TITLE
sysmenu: init at unstable-2024-06-13

### LINKED ISSUE #320369: Package request: sysmenu
**Project description**

Sysmenu is a simple and efficient application launcher written in gtkmm4
**Metadata**

* homepage URL: https://github.com/System64fumo/sysmenu
* source URL: https://github.com/System64fumo/sysmenu
* license: wtfpl
* platforms: linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by matteo-pacini**
## Description of changes

Sysmenu is a simple and efficient application launcher written in gtkmm4.

Fixes  #320369.

Feels like an `unstable` package, hence the choice of version and updater.

Changes from upstream:
- Removed runtime config - config can be done via program arguments
- Removed non-deterministic compile flags

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [X] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/sy/sysmenu/package.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+}:
+stdenv.mkDerivation {
+  pname = "sysmenu";
+  version = "unstable-13-06-2024";
+
+  src = fetchFromGitHub {
+    owner = "System64fumo";
+    repo = "sysmenu";
+    rev = "0b891e5d27f286b867d4a0984ee284a3456de851";
+    sha256 = "sha256-zI6dmS+ZqcGrG/joWouqrMNQfTRflP677kYJLJEWTBc=";
```
Comment (jtbx) [inline pkgs/by-name/sy/sysmenu/package.nix:20]: `hash` is preferred over sha256 for new packages
```suggestion
    hash = "sha256-zI6dmS+ZqcGrG/joWouqrMNQfTRflP677kYJLJEWTBc=";
```

File: pkgs/by-name/sy/sysmenu/package.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  installPhase = ''
+    runHook preInstall
+    install -Dm755 sysmenu $out/bin/sysmenu
+    runHook postInstall
+  '';
+
+  passthru.updateScript = unstableGitUpdater { };
+
+  meta = {
+    description = "Simple program launcher written in gtkmm 4";
```
Comment (jtbx) [inline pkgs/by-name/sy/sysmenu/package.nix:54]: I think the only thing that matters here is that this program uses GTK4
```suggestion
    description = "Simple program launcher using GTK4";
```

File: pkgs/by-name/sy/sysmenu/package.nix [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    homepage = "https://github.com/System64fumo/sysmenu";
+    license = lib.licenses.wtfpl;
+    mainProgram = "sysmenu";
+    maintainers = [ lib.maintainers.matteopacini ];
```
Comment (jtbx) [inline pkgs/by-name/sy/sysmenu/package.nix:58]:
```suggestion
    maintainers = with lib.maintainers; [ matteopacini ];
```

File: pkgs/by-name/sy/sysmenu/package.nix [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  buildInputs = [
+    gtkmm4
+    gtk4-layer-shell
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+    make sysmenu
+    runHook postBuild
+  '';
```
Comment (jtbx) [inline pkgs/by-name/sy/sysmenu/package.nix:43]: sysmenu is already the default target in this makefile. We can actually use stdenv's default builder here which already runs make, just remove this build phase

**Review by jtbx — CHANGES_REQUESTED** [review]
Got a few notes, apart from that it's all good

**Comment by matteo-pacini** [comment]
@jtbx thanks for feedback, ready for re-review!

**Review by jtbx — APPROVED** [review]
Awesome thanks for that. Looks good :+1:

File: pkgs/by-name/sy/sysmenu/package.nix [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  pkg-config,
+  git,
+  gtkmm4,
+  gtk4-layer-shell,
+  wrapGAppsHook4,
+  unstableGitUpdater,
+}:
+stdenv.mkDerivation {
+  pname = "sysmenu";
+  version = "unstable-13-06-2024";
```
Comment (Aleksanaa) [inline pkgs/by-name/sy/sysmenu/package.nix:14]:
```suggestion
  version = "unstable-2024-06-13";
```

**Review by Aleksanaa — COMMENTED** [review]
Better to reformat with `nixfmt-rfc-style`, as your format is a bit compact

File: pkgs/by-name/sy/sysmenu/package.nix (hunk 5 above)
Comment (Aleksanaa) [reply pkgs/by-name/sy/sysmenu/package.nix:14]: also commit message

Comment (matteo-pacini) [reply pkgs/by-name/sy/sysmenu/package.nix:14]: @Aleksanaa I'm using that formatting as far as I am aware?

image

**Comment by matteo-pacini** [comment]
@Aleksanaa @jtbx ready for re-review thanks!

**Review by Aleksanaa — APPROVED** [review]
Tested

**Pull request merged**